In [28]:
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score
from sklearn.datasets import load_breast_cancer
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier


data = load_breast_cancer()
df = pd.DataFrame(data.data, columns = data.feature_names)
df["target"] = data.target

X = df.drop("target", axis=1)
y = df["target"]

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state=50)

# Unscaled (like Day 10)
log_model = LogisticRegression(max_iter=5000)
log_model.fit(X_train, y_train)
unscaled_acc = accuracy_score(y_test, log_model.predict(X_test))

# Scaled
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

log_model_scaled = LogisticRegression(max_iter=5000)
log_model_scaled.fit(X_train_scaled, y_train)
scaled_acc = accuracy_score(y_test, log_model_scaled.predict(X_test_scaled))

print("Unscaled:", unscaled_acc)
print("Scaled:", scaled_acc)

Unscaled: 0.9736842105263158
Scaled: 1.0


In [30]:
model = RandomForestClassifier(random_state=50)
model.fit(X_train, y_train)
unscaled_acc_rfc = accuracy_score(y_test, model.predict(X_test))

model.fit(X_train_scaled, y_train)
scaled_acc_rfc = accuracy_score(y_test, log_model_scaled.predict(X_test_scaled))

print("Unscaled:", unscaled_acc_rfc)
print("Scaled:", scaled_acc_rfc)

Unscaled: 0.956140350877193
Scaled: 1.0


In [23]:
model = RandomForestClassifier(random_state=42)
model.fit(X_train, y_train)
unscaled_acc_rfc = accuracy_score(y_test, model.predict(X_test))

model.fit(X_train_scaled, y_train)
scaled_acc_rfc = accuracy_score(y_test, log_model_scaled.predict(X_test_scaled))

print("Unscaled:", unscaled_acc_rfc)
print("Scaled:", scaled_acc_rfc)

Unscaled: 0.9473684210526315
Scaled: 1.0


Testing multiple random_state values shows Random Forest's accuracy varies significantly (0.96–1.0) regardless of scaling — sometimes scaled is higher, sometimes unscaled is higher. This confirms scaling doesn't meaningfully affect Random Forest; the earlier apparent improvement was just random variation. Logistic Regression's improvement from scaling was consistent and real, unlike Random Forest's.